ezGroceriesShoppingList/src/test/java/com/ezgroceries/shoppinglist/WebControllerTests.java

```diff
@@ -0,0 +1,129 @@
+package com.ezgroceries.shoppinglist;
+
+import com.ezgroceries.shoppinglist.entity.Cocktail;
+import com.ezgroceries.shoppinglist.entity.ShoppingList;
+import com.ezgroceries.shoppinglist.service.CocktailServiceImpl;
+import com.ezgroceries.shoppinglist.service.ShoppingListService;
+import com.ezgroceries.shoppinglist.web.WebController;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.List;
+
+import static com.ezgroceries.shoppinglist.util.TestData.*;
+import static com.ezgroceries.shoppinglist.util.TestHelperMethods.resourceToString;
+import static org.hamcrest.Matchers.hasSize;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(WebController.class)
+@AutoConfigureMockMvc(addFilters = false)
+public class WebControllerTests {
+
+
+    @MockBean
+    CocktailServiceImpl cocktailService;
+
+    @MockBean
+    ShoppingListService shoppingListService;
+
+    @SuppressWarnings("SpringJavaInjectionPointsAutowiringInspection")
+    @Autowired
+    MockMvc mockMvc;
+
+    @Test
+    public void testListAllCocktails() throws Exception {
+        List<Cocktail> cocktailList = getCocktailList();
+
+        when(cocktailService.listCocktails(eq("test"))).thenReturn(cocktailList);
+
+        mockMvc.perform(get("/cocktails?search=test"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", hasSize(2)))
+                .andExpect(jsonPath("$[0].name").value("Margerita"));
+
+        Mockito.verify(cocktailService, times(1)).listCocktails("test");
+    }
+
+    //Ignore the method below. It doesn't do anything. I created it before I realized that there is no direct way to test this method by url
```

Comment:
Why is that

Reply:
Because this method can not be accessed by an endpoint. It is only used by another service method accessed by the endpoint to add a cocktail to a shopping list and the second one is already tested.